sentry-types/src/protocol/v7.rs

```diff
@@ -1648,3 +1648,82 @@ impl<'a> fmt::Display for Transaction<'a> {
         )
     }
 }
+
+/// The different types an attachment can have
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum AttachmentType {
+    /// A standard attachment without special meaning. (default)
+    Attachment,
+    /// A minidump file that creates an error event and is symbolicated. The
+    /// file should start with the `MDMP` magic bytes.
+    Minidump,
+    /// An Apple crash report file that creates an error event and is symbolicated.
+    AppleCrashReport,
+    /// An XML file containing UE4 crash meta data. During event ingestion,
+    /// event contexts and extra fields are extracted from this file.
+    UnrealContext,
+    /// A plain-text log file obtained from UE4 crashes. During event ingestion,
+    /// the last logs are extracted into event breadcrumbs.
+    UnrealLogs,
+}
+
+impl Default for AttachmentType {
+    fn default() -> Self {
+        Self::Attachment
+    }
+}
+
+impl AttachmentType {
+    /// Gets the string value Sentry expects for the attachment type
+    pub fn as_str(self) -> &'static str {
+        match self {
+            Self::Attachment => "event.attachment",
+            Self::Minidump => "event.minidump",
+            Self::AppleCrashReport => "event.applecrashreport",
+            Self::UnrealContext => "unreal.context",
+            Self::UnrealLogs => "unreal.logs",
+        }
+    }
+}
+
+#[derive(Clone, PartialEq)]
+/// Represents an attachment item
+pub struct Attachment {
+    /// The actual attachment data
+    pub buffer: std::sync::Arc<Vec<u8>>,
```

Comment:
I wonder if its worth doing the same `Cow` with `'static` bound as we have for the other envelope items.
Also, I will look this up real quick myself, but do we actually clone this somewhere internally that warrants using an `Arc` here?

Reply:
I think there is Clone derive on something somewhere, but I can try removing it and see what happens.

Reply:
So the reason I initially did that was because of the Clone requirement on Envelope/EnvelopeItem, but then I noticed that...those are never being used and just removed the Clone requirements altogether.